pkgs/tools/networking/croc/test-local-relay.nix

```diff
@@ -0,0 +1,20 @@
+{ stdenv, croc }:
+
+stdenv.mkDerivation {
+  name = "croc-test-local-relay";
+  meta.timeout = 10;
+  buildCommand = ''
+          HOME=$(mktemp -d)
+          # start a local relay
+          timeout --preserve-status 5s ${croc}/bin/croc relay --ports 11111,11112 &
```

Comment:
This timeout kills the relay after 5s. Is that necessary?

Reply:
It seems the relay is killed by nix at the end of the test. Given that `passthru.tests` is not documented in detail, I felt more comfortable to kill the relay explicitly to avoid a hanging/timeout in the test.
5s should be ample time to run this test even on a very slow CPU.

Reply:
Nix will kill any processes that remain, not wait for them. `passthru.test` isn't too special. It's just "built" like any other derivation.

Reply:
Convinced. Dropped the timeout here.